File: backend/src/RegresaACasa.Api/Models/Entities/Pet.cs
namespace RegresaACasa.Api.Models.Entities;

/// <summary>
/// Publicación de una mascota perdida (tabla PETS).
/// </summary>
public class Pet
{
    public Guid Id { get; set; }
    public string PetType { get; set; } = string.Empty;
    public string? Name { get; set; }
    public string? Breed { get; set; }
    public string ColorDescription { get; set; } = string.Empty;
    public string Zone { get; set; } = string.Empty;
    public string ContactInfo { get; set; } = string.Empty;
    public string ImageUrl { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
}

File: mobile/src/models/pet.ts
// MODELO: tipos que reflejan exactamente el contrato JSON de la API (snake_case).
// Ver docs/api-contract.md.

export type Pet = {
  id: string;
  pet_type: string;
  name: string | null;
  breed: string | null;
  color_description: string;
  zone: string;
  contact_info: string;
  image_url: string; // URL firmada de lectura (caduca); no guardarla
};

export type CreatePetInput = {
  pet_type: string;
  name?: string;
  breed?: string;
  color_description: string;
  zone: string;
  contact_info: string;
  image_url: string;
};

export const PET_TYPES = ['Perro', 'Gato', 'Otro'] as const;

/** Reglas de validación del cuestionario (espejo de CreatePetRequest en el backend). */
export function validatePetForm(input: Omit<CreatePetInput, 'image_url'>, hasImage: boolean): string | null {
  if (!input.pet_type.trim()) return "El campo 'pet_type' es requerido";
  if (!input.color_description.trim()) return "El campo 'color_description' es requerido";
  if (!input.zone.trim()) return "El campo 'zone' es requerido";
  if (!input.contact_info.trim()) return "El campo 'contact_info' es requerido";
  if (!hasImage) return 'Agrega una foto de la mascota';
  return null;
}
